### PR TITLE
Read location data from csv file and plot markers on map. Info screens open on click.

### PR DESCRIPTION
**Pull request by margaretchan**
Primary: Derrill

Screenshot of map features: OGTQZ2n74Wz

### CODE REVIEW

**Comment by margaretchan** [comment]
Was unable to make the ImmutableList final and initialize in constructor. I got this error b7Q1BMJqBNK when I tried to do so.

File: portfolio/src/main/java/com/google/sps/servlets/LocationDataServlet.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    
+    private static final String CSV_FILE_PATH = "/WEB-INF/location-data-ithaca.csv";
+
+    // This list is thread-safe since it is only written to once in init() when the servlet is created.
+    // All subsequent reads can only occour after the init() had completed execution.
+    private ImmutableList<Location> locations;
+
+    @Override
+    public void init () {
+        ImmutableList.Builder builder = new ImmutableList.Builder<Location>();
```
Comment (derrillhd) [inline portfolio/src/main/java/com/google/sps/servlets/LocationDataServlet.java:39]: Don't use raw types.

ImmutableList.Builder<Location> builder = new ImmutableList.Builder<>();

An IDE  would have caught this.

File: portfolio/src/main/java/com/google/sps/servlets/LocationDataServlet.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        Scanner scanner = new Scanner(getServletContext().getResourceAsStream("/WEB-INF/location-data-ithaca.csv"));
+        
+        while (scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+            String[] values = line.split(",");
+            builder.add(new Location(Double.parseDouble(values[0]), Double.parseDouble(values[1]), values[2], values[3]));
```
Comment (derrillhd) [inline portfolio/src/main/java/com/google/sps/servlets/LocationDataServlet.java:45]: optional.  for readability you could assign each one of these arguments as a variable.. then it'll read:


new Location(latitude, longitude, placeName, description) (for example)

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -12,6 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+const MAP_INIT_LAT = 42.446263;
```
Comment (derrillhd) [inline portfolio/src/main/webapp/script.js:15]: document where this is.

File: portfolio/src/main/java/com/google/sps/servlets/LocationDataServlet.java (hunk 1 above)
Comment (margaretchan) [reply portfolio/src/main/java/com/google/sps/servlets/LocationDataServlet.java:39]: Changed.

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (margaretchan) [reply portfolio/src/main/webapp/script.js:15]: Added comment.

File: portfolio/src/main/java/com/google/sps/servlets/LocationDataServlet.java (hunk 2 above)
Comment (margaretchan) [reply portfolio/src/main/java/com/google/sps/servlets/LocationDataServlet.java:45]: Added variables for each parameter.

**Pull request merged**